docs/spec.md

```diff
@@ -116,6 +116,68 @@ types, for example, to include layouts
 ([#629](https://github.com/openxla/stablehlo/issues/629)) and sparsity
 ([#1078](https://github.com/openxla/stablehlo/issues/1078)).
 
+```ebnf
+QuantizedTensorType ::= 'tensor' '<' TensorShape QuantizedElementType '>'
+QuantizedElementType ::= '!quant.uniform' '<'
+                  QuantizationStorageType
+                  ['<' QuantizationStorageMin ':' QuantizationStorageMax '>']
+                  ':' QuantizationExpressedType
+                  [':' QuantizationDimension]
+                  ',' QuantizationParameters '>'
+QuantizationStorageType ::= IntegerType
+QuantizationStorageMin ::= IntegerConstant
+QuantizationStorageMax ::= IntegerConstant
+QuantizationExpressedType ::= FloatType
+QuantizationDimension ::= IntegerConstant
+QuantizationParameters ::= QuantizationParameter
+                         | '{' QuantizationParameter {',' QuantizationParameter} '}'
+QuantizationParameter ::= QuantizationScale ':' QuantizationZeroPoint
+QuantizationScale ::= FloatConstant
+QuantizationZeroPoint ::=  IntegerConstant
+```
+
+**Quantized element types** represent integer values of a **storage type** in
+the range from `storage_min` to `storage_max` (inclusive) that correspond to
+floating-point values of an **expressed type**. For a given integer value `i`,
+the corresponding floating-point value `f` can be computed as
+`f = (i - zero_point) * scale`, where `scale` and `zero_point` are called
+**quantization parameters**. The `storage_min` and `storage_max` are optional
+in the grammar, but have default values of `min_value(storage_type)` and
+`max_value(storage_type)` respectively. Quantized element types have the
+following constraints:
+
+* (C1) `num_bits(storage_type) < num_bits(expressed_type)`.
+* (C2) `type(storage_min) = storage_type`.
+* (C3) `type(storage_max) = storage_type`.
+* (C4) `min_value(storage_type) <= storage_min < storage_max <= max_value(storage_type)`.
+* (C5) For all `i`, `type(scales[i]) = expressed_type`.
+* (C6) For all `i`, `scales[i] > 0`.
+* (C7) For all `i`, `storage_min <= zero_points[i] <= storage_max`.
```

Comment:
In TFLite `zero_point` is per-tensor, even when per-channel quantization is used ("per-channel" affects only scales, and BTW per-channel quantization is supported only for signed quantization schema, where zero point must be 0).

Reply:
Thanks @Maratyszcza for the valuable information! 
The constraint we have is straight from text-book definition of per-axis zero points and hence is more general. But I am open to improvement.  IMO, this is also somewhat related to our prior discussion about the trade-off between generality of spec vs practical values of zero points.  [here](https://github.com/openxla/stablehlo/pull/1352#discussion_r1147883143). 


@sngyhan Do you mind commenting on this in the light of the following:

For dynamic range quantization, for say convolution op, when the activation is quantized on the fly, I can see from [here](https://github.com/tensorflow/tensorflow/blob/3a838bf19754eccfc7a9d324ba0e874735a77383/tensorflow/core/kernels/uniform_quant_ops/uniform_quantized_convolution_ops.cc#L717) and [here](https://github.com/tensorflow/tensorflow/blob/3a838bf19754eccfc7a9d324ba0e874735a77383/tensorflow/core/kernels/uniform_quant_ops/math_utils.h#L165), that the activation quantization is per-channel (on batch dimension) but is asymmetric with zero point  != 0. Moreover, IIUC, in most of the kernel implementations in https://github.com/tensorflow/tensorflow/tree/3a838bf19754eccfc7a9d324ba0e874735a77383/tensorflow/core/kernels/uniform_quant_ops are using per-channel zero-points and they does not seem to have special value constraint on them.

Can you please let me know if  I missing something.

Reply:
Similarly to other unresolved conversation, let's follow up on this in a dedicated ticket: #1405.